boot/modeenv.go

```diff
@@ -111,6 +121,34 @@ func ReadModeenv(rootdir string) (*Modeenv, error) {
 	unmarshalModeenvValueFromCfg(cfg, "current_trusted_boot_assets", &m.CurrentTrustedBootAssets)
 	unmarshalModeenvValueFromCfg(cfg, "current_trusted_recovery_boot_assets", &m.CurrentTrustedRecoveryBootAssets)
 
+	// save all the rest of the keys we don't understand
+	keys, err := cfg.Options("")
+	if err != nil {
+		return nil, err
+	}
+	var knownKeys = []string{
+		"recovery_system",
+		"current_recovery_systems",
+		"mode",
+		"base",
+		"base_status",
+		"try_base",
+		"current_kernels",
+		"model",
+		"grade",
+		"current_trusted_boot_assets",
+		"current_trusted_recovery_boot_assets",
+	}
+	for _, k := range keys {
+		if !strutil.ListContains(knownKeys, k) {
```

Comment:
We have a bunch of fields, so tweaked this to use `map[string]struct{}`, rather than scanning the array for each field.